client/src/partials/templates/modals/location.modal.html

```diff
@@ -0,0 +1,135 @@
+    <div class="form-group"
+      ng-class="{ 'has-error' : LocationModalForm.$submitted && LocationModalForm.country.$error }"
+      >
+      <label class="control-label">
+        {{ "LOCATION.COUNTRY" | translate }}
+      </label>
+      <select
+        ng-if="LocationModalCtrl.view.index > 1"
+        class="form-control"
+        name="country"
+        ng-model="LocationModalCtrl.country"
+        ng-change="LocationModalCtrl.loadProvinces()"
+        ng-options="country as country.name for country in LocationModalCtrl.countries track by country.uuid"
+        required
+        >
+        <option value="" disabled>{{ LocationModalCtrl.messages.country | translate }}</option>
+      </select>
+      <input
+        ng-if="LocationModalCtrl.view.index === 1"
+        class="form-control"
+        name="country"
+        ng-model="LocationModalCtrl.country"
+        placeholder="{{ 'INPUT.COUNTRY' | translate }}"
+        required>
+    </div>
+
+    <div class="form-group"
+      ng-class="{ 'has-error' : LocationModalForm.$submitted && LocationModalForm.province.$error }"
+      ng-if="LocationModalCtrl.view.index > 1"
+      >
+      <label class="control-label">
+        {{ "LOCATION.PROVINCE" | translate }}
+      </label>
+      <select
+        ng-if="LocationModalCtrl.view.index > 2"
+        class="form-control"
+        name="province"
+        ng-model="LocationModalCtrl.province"
+        ng-options="province as province.name for province in LocationModalCtrl.provinces track by province.uuid"
+        ng-change="LocationModalCtrl.loadSectors()"
+        required>
+        <option value="" disabled>{{ LocationModalCtrl.messages.province | translate }}</option>
+      </select>
+
+      <input
+        ng-if="LocationModalCtrl.view.index === 2"
+        class="form-control"
+        name="province"
+        ng-model="LocationModalCtrl.province"
+        placeholder="{{ 'INPUT.PROVINCE' | translate }}"
+        required>
+    </div>
+
+    <div class="form-group"
+      ng-if="LocationModalCtrl.view.index > 2"
+      ng-class="{ 'has-error' : LocationModalForm.$submitted && LocationModalForm.sector.$error }"
+      >
+      <label class="control-label">
+        {{ "LOCATION.SECTOR" | translate }}
+      </label>
+      <select
+        ng-if="LocationModalCtrl.view.index > 3"
+        class="form-control"
+        name="sector"
+        ng-model="LocationModalCtrl.sector"
+        ng-options="sector as sector.name for sector in LocationModalCtrl.sectors track by sector.uuid"
+        required>
+        <option value="" disabled>{{ LocationModalCtrl.messages.sector | translate }}</option>
+      </select>
+      <input
+        ng-if="LocationModalCtrl.view.index === 3"
+        name="sector"
+        class="form-control"
+        ng-model="LocationModalCtrl.sector"
+        placeholder="{{ 'INPUT.SECTOR' | translate }}"
+        required>
+    </div>
+
+    <div class="form-group"
+      ng-if="LocationModalCtrl.view.index > 3"
```

Comment:
For the hard-coded part, these are hard-coded in the controller and will not change - they are simply a visual hierarchy for the view.

For the numbers, you are right, and I tried for a long time to find a way that didn't depend on numbers.  Unfortunately, there is no way to capture the dependency relationship between `village`, `sector`, `province`, and `country`.  If you can think of a better way, I'd be happy to change it.